src/resource_manager/gateway.py

```diff
@@ -27,8 +26,26 @@
 GATEWAY_PLURAL = "gateways"
 
 
-class CreateGatewayError(Exception):
-    """Represents an error when creating the gateway resource."""
+@dataclasses.dataclass
+class GatewayState(State):
```

Comment:
As mentioned in another comment, we should probably rename this to `GatewayResourceDefinition`

Reply:
That makes sense, I've renamed the class to GatewayResourceDefinition, the component to GatewayResourceInformation in all relevant places.